src/bootstrap/builder.rs

```diff
@@ -1235,6 +1235,21 @@ impl<'a> Builder<'a> {
             compiler.stage
         };
 
+        // Export target-specific RUSTFLAGS to Cargo.  We must include flags
+        // for both the target and the host in case there are proc-macros.
+        let mut export_target_rustflags = |target| {
+            if let Some(target_cfg) = self.config.target_config.get(&target) {
+                if let Some(flags) = &target_cfg.rustflags {
+                    cargo.env(format!("CARGO_TARGET_{}_RUSTFLAGS", target.triple), flags);
```

Comment:
https://doc.rust-lang.org/nightly/cargo/reference/config.html#buildrustflags implies that cargo will not merge these with RUSTFLAGS, which we also set...

Does this PR work? What am I missing?